homeassistant/components/rest/data.py

```diff
@@ -63,7 +63,7 @@ async def async_update(self, log_errors=True):
                 headers=rendered_headers,
                 params=rendered_params,
                 auth=self._auth,
-                data=self._request_data,
+                content=self._request_data,
```

Comment:
cc @bdraco

I think this was missed in #41973, during the migration to `httpx`

Reply:
https://www.python-httpx.org/api/
```
content - (optional) Binary content to include in the body of the request, as bytes or a byte iterator.
data - (optional) Form data to include in the body of the request, as a dictionary.
```

Reply:
Yes, but `CONF_PAYLOAD` is defined as string.
https://github.com/home-assistant/core/blob/76bd8c431b7e45ec3c5fc872d3d36736534f0433/homeassistant/components/rest/schema.py#L57

Reply:
`payload string (optional)
The payload to send with a POST request. Depends on the service, but usually formed as JSON.`

So it should be `content` since its not a `dict` for `rest`